linker-interface/shared/src/main/scala/org/scalajs/linker/interface/LinkTimeProperties.scala

```diff
@@ -0,0 +1,64 @@
+package org.scalajs.linker.interface
+
+import org.scalajs.ir.Trees._
+
+final class LinkTimeProperties (
+  semantics: Semantics,
+  esFeatures: ESFeatures
+) {
+  import LinkTimeProperties._
+  import LinkTimeProperties.ResolvedLinkTimeValue._
+
+  private val info = "scala.scalajs.LinkingInfo"
+  private val linkTimeProperties: Map[String, ResolvedLinkTimeValue] = Map(
+    s"$info.productionMode" -> BooleanValue(semantics.productionMode),
+    s"$info.esVersion"      -> IntValue(esFeatures.esVersion.edition)
```

Comment:
I would recommend `"core/productionMode"` and `"core/esVersion"` here. We shouldn't refer to the `scala.*` namespace in the core IR spec.